lib/logstash/plugin_mixins/elasticsearch/common.rb

```diff
@@ -192,6 +192,26 @@ def retrying_submit(actions)
       end
     end
 
+    # we shouldn't unsupported actions to ES
+    # method filters out unsupported actions by warning, writes event to DQL if enabled
+    # @returns valid actions
+    def filter_unsupported_actions(actions)
+      return if actions.empty? || actions.size < 1
+      grouped_actions = actions.group_by { |action, arg, source | LogStash::Outputs::ElasticSearch::VALID_HTTP_ACTIONS.include?(action) }
+      unless grouped_actions[false].nil?
+        @logger.warn("Number of requests filtered out before sending to Elasticsearch because of unsupported action, ", size: grouped_actions[false].size)
+        write_unsupported_events_to_dql(grouped_actions[false])
+      end
+      grouped_actions[true].nil? ? [] : grouped_actions[true]
+    end
```

Comment:
I would recommend using `Enumerable#partition` instead of `Enumerable#group_by` since the result can be bound to two local variables that can be handled directly, instead of digging into a hash repeatedly.

```suggestion
    def filter_unsupported_actions(actions)
      supported_actions, unsupported_actions = actions.partition { |action, _, _| VALID_HTTP_ACTIONS.include?(action) }
      
      handle_dlq_or_drop(unsupported_actions)
      
      supported_actions
    end
    
    def handle_dlq_or_drop(unsupported_actions)
      return if unsupported_actions.nil? || unsupported_actions.empty?
      
      # ... implement routing to dlq or drop.
      # logging in data-loss situations should mirror our existing dropped events logging in detail.
    end
```

Reply:
Not sure why `VALID_HTTP_ACTIONS` is not injected, getting following error:
```
[ERROR][logstash.javapipeline    ][main] Pipeline worker error, the pipeline will be stopped {:pipeline_id=>"main",
 :error=>"(NameError) uninitialized constant LogStash::PluginMixins::ElasticSearch::Common::VALID_HTTP_ACTIONS", :exception=>Java::OrgJrubyExceptions::NameError,
```